components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java

```diff
@@ -266,6 +271,41 @@ private void validateClaimValueForRegex(String claimURI, String claimValue, Stri
         }
     }
 
+    /**
+     * Validate attribute values against length limits.
+     *
+     * @param claimURI      Claim URI.
+     * @param value         Claim value.
+     * @param tenantDomain  Tenant domain name..
+     * @throws UserStoreClientException If an error occurred in validating claim.
+     */
+    private void validateLength(String claimURI, String value, String tenantDomain) throws UserStoreClientException {
+
+        if (StringUtils.isBlank(claimURI)) {
+            if (log.isDebugEnabled()) {
+                log.debug("The claim URI is empty.");
+            }
+            return;
+        }
+        Map<String, String> claimProperties = getClaimProperties(tenantDomain, claimURI);
+        if (MapUtils.isEmpty(claimProperties)) {
+            return;
+        }
+        String minLength = claimProperties.get(MIN_LENGTH);
+        String maxLength = claimProperties.get(MAX_LENGTH);
+        boolean required = Boolean.parseBoolean(claimProperties.get(REQUIRED));
+
+        if (!required && StringUtils.isBlank(value)) {
+            return;
+        }
+        if ((StringUtils.isNotEmpty(minLength) && Integer.parseInt(minLength) > value.length()) ||
```

Comment:
Shall we use isNotBlank here ? What happens if there is a parsing exception due to invalid characters?

Reply:
Fixed with c47a013af43b8e50ada82cdecdf6a275bc31f2b0